pkg/cmd/cli/cmd/set/env.go

```diff
@@ -275,10 +346,29 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 				if list {
 					fmt.Fprintf(out, "# %s %s, container %s\n", info.Mapping.Resource, info.Name, c.Name)
 					for _, env := range c.Env {
-						// if env.ValueFrom != nil && env.ValueFrom.FieldRef != nil {
-						// 	fmt.Fprintf(cmd.Out(), "%s= # calculated from pod %s %s\n", env.Name, env.ValueFrom.FieldRef.FieldPath, env.ValueFrom.FieldRef.APIVersion)
-						// 	continue
-						// }
+						if resolve && env.ValueFrom != nil {
+							if env.ValueFrom.ConfigMapKeyRef != nil {
+								env.Value, err = getConfigMapRefValue(f, env.ValueFrom.ConfigMapKeyRef)
+								if err != nil {
+									fmt.Printf("error retrieving configmap: #%v\n", err)
```

Comment:
Otherwise would be very awkward to an end user whether the value is really empty or is a ref that would require to issue the "oc env" with --resolve.